### PR TITLE
Jep-200 test failure fix

### PR DESCRIPTION
**Pull request by RobinRSchulz**
- Fix failing tests
- Avoid using XSTREAM with mocked classes.
- Update the touched tests to use JenkinsRule instead of mocks.

### CODE REVIEW

File: src/test/java/hudson/plugins/jobConfigHistory/JobConfigBadgeActionTest.java [hunk 1]
```diff
@@ -23,30 +23,36 @@
...
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.GregorianCalendar;
 import java.util.List;
+import java.util.SortedMap;
 
+import hudson.model.*;
```
Comment (stefanbrausch) [inline src/test/java/hudson/plugins/jobConfigHistory/JobConfigBadgeActionTest.java]: Please no * imports

File: src/test/java/hudson/plugins/jobConfigHistory/JobConfigHistoryRootActionTest.java [hunk 2]
```diff
@@ -23,31 +23,32 @@
  */
 package hudson.plugins.jobConfigHistory;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (stefanbrausch) [inline src/test/java/hudson/plugins/jobConfigHistory/JobConfigHistoryRootActionTest.java]: No * imports

File: src/test/java/hudson/plugins/jobConfigHistory/JobConfigHistoryTest.java [hunk 3]
```diff
@@ -39,38 +39,38 @@
 
 import javax.servlet.ServletException;
 
+import hudson.model.*;
```
Comment (stefanbrausch) [inline src/test/java/hudson/plugins/jobConfigHistory/JobConfigHistoryTest.java]: * import

File: src/test/java/hudson/plugins/jobConfigHistory/JobConfigBadgeActionTest.java (hunk 1 above)
Comment (RobinRSchulz) [reply src/test/java/hudson/plugins/jobConfigHistory/JobConfigBadgeActionTest.java]: Done

**Pull request merged**